src/main/java/org/mule/extension/oauth2/internal/AbstractGrantType.java

```diff
@@ -303,4 +306,15 @@ public boolean isDefaultTokenManager() {
     return defaultTokenManager || tokenManager == null;
   }
 
+  @Override
+  public boolean readsAuthenticatedResponseBody() {
+    if (readsResponseBody == null) {
```

Comment:
put this initialization login in `initialize`, not here. In any way, this code is not thread safe

Reply:
Moved to initialize